config/m4/compiler.m4

```diff
@@ -563,6 +567,16 @@ ADD_COMPILER_FLAGS_IF_SUPPORTED([[-Wno-missing-field-initializers],
                                  [-Winvalid-pch]],
                                 [AC_LANG_SOURCE([[int main(int argc, char **argv){return 0;}]])])
 
+#
+# Intel Compiler specifics
+#
+if ! $CC -V | grep -q Intel.*Compiler; then
```

Comment:
why need to check for intel compiler - I think ADD_COMPILER_FLAGS_IF_SUPPORTED would not add unsupported flags anyway?

Reply:
yes we could, but since they relax code checking constraints, i thought to do it in a selective way since the failure is specific to intel compiler behavior i think:
1. this one is to be able to build with inline
2. this one is to address reported issue #9888 
3. this is for allow isnan() which is a macro in math.h skipped with icx compiler that uses empty parameter

Reply:
Not sure I understand the explanation.. do you mean ADD_COMPILER_FLAGS_IF_SUPPORTED will not always work correctly for non-intel compilers?

Reply:
As you suggested, simply adding to existing list, without checking for icc first, would work I think. But since it relaxes the compiler checks, it could lead to less strict compiling warning/errors.

Moreover the reason for relaxing those requirements appear to be very icc/icx specific behavior, for instance:
- Wno-language-extension-token is needed otherwise `inline` is not accepted (but __typeof__ etc become allowed).
- -fno-finite-math-only is needed otherwise isnan isinf is producing error on icx, other compilers simply replace those by true when applicable
- -Wno-recommended-options seems to be icx only anyways (recommends -fopenmp -> -qopenmp)
- Wno-c99-extentions, math.h shipped with icx defines isnan() as a macro with an empty argument which triggers an error

Reply:
so maybe need a AC_LANG_SOURCE style test to test if each thing is needed?
re isnan/isinf - maybe we can fix it in the code?

Reply:
could be done, but since every compiler/version will have its own needs, i think it might be more maintainable to simply add as is.

for isnan/isinf, i thought about this. there is `__FINITE_MATH_ONLY__ ` but icx does not define it...

Reply:
ok